components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php

```diff
@@ -0,0 +1,377 @@
+{
+    private ilCtrlInterface $ctrl;
+    private ilLanguage $lng;
+    private ilGlobalTemplateInterface $tpl;
+    private ilTabsGUI $tabs;
+    private ilRbacSystem $rbacsystem;
+    private ilSetting $setting;
+    private ilErrorHandling $ilErr;
+    private ?ilPropertyFormGUI $form;
+    private \ILIAS\HTTP\Services $http;
+    private \ILIAS\Refinery\Factory $refinery;
+    private \ILIAS\UI\Factory $ui_factory;
+    private \ILIAS\UI\Renderer $ui_renderer;
+    private int $ref_id;
+    private ilAuthLogoutPageEditorSettings $settings;
+    private ?ilSetting $logout_settings = null;
+    private \ILIAS\Style\Content\Object\ObjectFacade $content_style_domain;
+    private ?string $redirect_source = null;
+    private ?int $key = null;
+
+    public function __construct(int $a_ref_id)
+    {
+        global $DIC;
+
+        $this->ctrl = $DIC->ctrl();
+        $this->tpl = $DIC->ui()->mainTemplate();
+        $this->tabs = $DIC->tabs();
+        $this->rbacsystem = $DIC->rbac()->system();
+        $this->setting = $DIC->settings();
+        $this->ilErr = $DIC['ilErr'];
+
+        $this->http = $DIC->http();
+        $this->refinery = $DIC->refinery();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->ui_renderer = $DIC->ui()->renderer();
+
+        $this->lng = $DIC['lng'];
+
+        $this->lng->loadLanguageModule('auth');
+        $this->ref_id = $a_ref_id;
+
+        $this->settings = ilAuthLogoutPageEditorSettings::getInstance();
+        $this->content_style_domain = $DIC->contentStyle()
+                                          ->domain()
+                                          ->styleForRefId($a_ref_id);
+
+        $query_wrapper = $DIC->http()->wrapper()->query();
+        $post_wrapper = $DIC->http()->wrapper()->post();
+        $refinery = $DIC->refinery();
+
+        if ($query_wrapper->has('redirectSource')) {
+            $this->redirect_source = $query_wrapper->retrieve('redirectSource', $refinery->kindlyTo()->string());
+        }
+
+        if ($post_wrapper->has('key')) {
+            $this->key = $post_wrapper->retrieve('key', $refinery->kindlyTo()->int());
+        } elseif ($query_wrapper->has('key')) {
+            $this->key = $query_wrapper->retrieve('key', $refinery->kindlyTo()->int());
+        }
+    }
+
+    public function getSettings(): ilAuthLogoutPageEditorSettings
+    {
+        return $this->settings;
+    }
+
+    public function getRefId(): int
+    {
+        return $this->ref_id;
+    }
+
+    public function executeCommand(): void
+    {
+        switch (strtolower($this->ctrl->getNextClass($this))) {
+            case strtolower(ilLogoutPageGUI::class):
+                $this->tabs->clearTargets();
+                $this->tabs->setBackTarget(
+                    $this->lng->txt('back'),
+                    $this->ctrl->getLinkTarget($this, 'show')
+                );
+
+                if ($this->redirect_source !== strtolower(ilInternalLinkGUI::class)) {
```

Comment:
Do we need to apply a `strtolower` to the left operand as well?

Reply:
yes, that makes sense